lib/inspec/resources/file.rb

```diff
@@ -165,7 +165,11 @@ def more_permissive_than?(max_mode = nil)
     end
 
     def to_s
-      "File #{source_path}"
+      if file
+        "File #{source_path}"
+      else
+        "Bad File on %s" % [inspec.backend.class]
```

Comment:
This set of lines is the important code in this PR; there is also some related test code. Most of the rest of the code delta is a test file refactor.

Reply:
Yes, as described verbosely in the commits.